### DISCUSSION TITLE
set_hook not available with `luau` feature?

### DISCUSSION THREAD

**Discussion by jqnatividad** [mlua-rs/mlua#227] [Q&A] [answered]
First off @khvzak, thanks for this wonderful crate and supporting luau!

However, I'm trying to use debugging with Luau but it seems `set_hook` is not available with the `luau` feature.

I noticed that hook.rs is not compiled with the `luau` feature, so is there an alternative way to do debugging with Luau?

**Comment by khvzak** [comment, marked as answer]
`set_hook` indeed is not available for Luau, because Luau (for performance reasons) does not support it :)
More details here: https://luau-lang.org/performance#epsilon-overhead-debugger

Instead, mlua supports [interrupts](https://docs.rs/mlua/0.8.6/mlua/struct.Lua.html#method.set_interrupt) which is Luau only feature, similar to hooks.
Currently callback function does not has `&Lua` argument but I plan to add it on the v0.9 release.

**Reply by jqnatividad** [reply]
Awesome, I'll experiment with interrupts and see if it works for my project! 

And I'm looking forward to 0.9!

BTW, do you have a rough ETA for it?

**Reply by khvzak** [reply]
>  do you have a rough ETA for it?

I hope end of December / beginning of January.
